gobblin-utility/src/test/java/org/apache/gobblin/util/AvroUtilsTest.java

```diff
@@ -310,4 +320,157 @@ public void testUnionWithNull() {
+    Assert.assertNotNull(outputRecord.getField("newField"));
+    Assert.assertEquals(outputRecord.getField("newField").schema(), fieldSchema);
+  }
+
+
+  @Test
+  public void testDecorateSchemaWithStringProperties() {
+
+    Schema inputRecord = SchemaBuilder.record("test").fields()
+            .name("integer1")
+              .prop("innerProp", "innerVal")
+              .type().intBuilder().endInt().noDefault()
+            .requiredString("string1")
+            .endRecord();
+    inputRecord.addProp("topLevelProp", "topLevelVal");
+    Schema.Field additionalField = getTestInnerRecordField();
+    Schema outputSchema = AvroUtils.decorateRecordSchema(inputRecord, Collections.singletonList(additionalField));
+    Assert.assertEquals(outputSchema.getProp("topLevelProp"), "topLevelVal");
+    Assert.assertEquals(outputSchema.getField("integer1").getProp("innerProp"), "innerVal");
+  }
+
+  @Test
+  public void testDecorateSchemaWithObjectProperties() throws IOException {
+
+    String customPropertyString = "{\"custom\": {\"prop1\": \"val1\"}}";
+    JsonNode customPropertyValue = new ObjectMapper().readTree(customPropertyString);
+
+    Schema inputRecord = SchemaBuilder.record("test").fields()
+            .name("integer1")
+            .prop("innerProp", "innerVal")
+            .type().intBuilder().endInt().noDefault()
+            .requiredString("string1")
+            .endRecord();
+    inputRecord.addProp("topLevelProp", customPropertyValue);
+    Schema.Field additionalField = getTestInnerRecordField();
+    Schema outputSchema = AvroUtils.decorateRecordSchema(inputRecord, Collections.singletonList(additionalField));
+    Assert.assertEquals(outputSchema.getProp("topLevelProp"), inputRecord.getProp("topLevelProp"));
+    Assert.assertEquals(outputSchema.getField("integer1").getProp("innerProp"), "innerVal");
+  }
+
+
+  private Schema.Field getTestInnerRecordField() {
+    Schema fieldSchema = SchemaBuilder.record("innerRecord")
+            .fields().requiredInt("innerInt").requiredString("innerString")
+            .endRecord();
+    Schema.Field field = new Schema.Field("innerRecord", fieldSchema, "",null);
+    return field;
+  }
+
+
+  @Test
+  public void testDecorateSchemaWithSingleRecord() {
+    Schema inputRecord = SchemaBuilder.record("test").fields().requiredInt("numeric1")
+            .requiredString("string1").endRecord();
+    Schema fieldSchema = SchemaBuilder.record("innerRecord")
+            .fields().requiredInt("innerInt").requiredString("innerString")
+            .endRecord();
+    Schema.Field field = new Schema.Field("innerRecord", fieldSchema, "",null);
+    Schema outputRecord = AvroUtils.decorateRecordSchema(inputRecord, Collections.singletonList(field));
+    Assert.assertNotNull(outputRecord.getField("innerRecord"));
+    Assert.assertEquals(outputRecord.getField("innerRecord").schema(), fieldSchema);
+  }
+
+
+  @Test
+  public void testDecorateRecordWithPrimitiveField() {
```

Comment:
Do you need this test, given the testDecorateRecordWithNestedField?

Reply:
Same reason as above for schemas.